tests/sensors/test_external_task_sensor.py

```diff
@@ -838,6 +839,172 @@ def test_external_task_group_when_there_is_no_TIs(self):
                 ignore_ti_state=True,
             )
 
+    @pytest.mark.parametrize(
+        "soft_fail, expected_exception, kwargs, expected_message",
+        (
+            (
+                False,
+                AirflowException,
+                {
+                    "external_task_ids": [TEST_TASK_ID, TEST_TASK_ID_ALTERNATE],
+                    "failed_states": [State.FAILED],
+                },
+                f"Some of the external tasks {re.escape(str([TEST_TASK_ID, TEST_TASK_ID_ALTERNATE]))}"
+                f" in DAG {TEST_DAG_ID} failed.",
+            ),
+            (
+                False,
+                AirflowException,
+                {
+                    "external_task_group_id": [TEST_TASK_ID, TEST_TASK_ID_ALTERNATE],
+                    "failed_states": [State.FAILED],
+                },
+                f"The external task_group '{re.escape(str([TEST_TASK_ID, TEST_TASK_ID_ALTERNATE]))}'"
+                f" in DAG '{TEST_DAG_ID}' failed.",
+            ),
+            (
+                False,
+                AirflowException,
+                {"failed_states": [State.FAILED]},
+                f"The external DAG {TEST_DAG_ID} failed.",
+            ),
+            (
+                True,
+                AirflowSkipException,
+                {
+                    "external_task_ids": [TEST_TASK_ID, TEST_TASK_ID_ALTERNATE],
+                    "failed_states": [State.FAILED],
+                },
+                "Skipping due to soft_fail is set to True.",
+            ),
+            (
+                True,
+                AirflowSkipException,
+                {
+                    "external_task_group_id": [TEST_TASK_ID, TEST_TASK_ID_ALTERNATE],
+                    "failed_states": [State.FAILED],
+                },
+                "Skipping due to soft_fail is set to True.",
+            ),
+            (
+                True,
+                AirflowSkipException,
+                {"failed_states": [State.FAILED]},
+                "Skipping due to soft_fail is set to True.",
+            ),
+        ),
+    )
+    @mock.patch("airflow.sensors.external_task.ExternalTaskSensor.get_count")
+    @mock.patch("airflow.sensors.external_task.ExternalTaskSensor._get_dttm_filter")
+    def test_fail_poke(
+        self, _get_dttm_filter, get_count, soft_fail, expected_exception, kwargs, expected_message
+    ):
+        _get_dttm_filter.return_value = []
+        get_count.return_value = 1
+        op = ExternalTaskSensor(
+            task_id="test_external_task_duplicate_task_ids",
+            external_dag_id=TEST_DAG_ID,
+            allowed_states=["success"],
+            dag=self.dag,
+            soft_fail=soft_fail,
+            deferrable=False,
+            **kwargs,
+        )
+        with pytest.raises(expected_exception, match=expected_message):
+            op.execute(context={})
+
+    @pytest.mark.parametrize(
```

Comment:
I might try something like suggested above.

Reply:
In this testcase, it might be a little difficult because the error message is coming from a location. tightly coupling `soft_fail`, `AirflowSkipException`, `kwargs` and `expected_message`

Reply:
We can add a one-line logic in the test case, like https://github.com/apache/airflow/pull/34652#discussion_r1340228280
But I'm actually ok with it. Just a minor suggestion :)

Reply:
Oh yeah, yup, that works. Now I'm a little concerned that it might be overly complicated. Nevertheless, I have added the change to make the testcase compact and elegant.